R/item_upload_files.R

```diff
@@ -0,0 +1,81 @@
+#' #' Upload file(s) and create a new item
+#' 
+#' Create a new item with files attached, all in one call to SB
+#' 
+#' @template item_with_parent
+#'
+#' @param files A string vector of paths to files to be uploaded
+#' @export
+#' @return An object of class \code{sbitem}
+#' @examples \dontrun{
+#' # You'll need a parent id for a folder/item
+#' ## here, using your highest level parent folder
+#' file <- system.file("examples", "books.json", package = "sbtools")
+#' item_upload_create(user_id(), file)
+#' }
+item_upload_create = function(parent_id, files, ..., session=current_session()){
+	
+	if(length(files) > 50){
+		warning('Trying to attach a large number of files to a SB item. SB imposes file limits which may cause this to fail')
+	}
+	
+	item <- as.sbitem(parent_id)
+	r = sbtools_POST(url = paste0(pkg.env$url_upload_create, item$id, '?title=title'), 
+									 ...,
+									 body = multi_file_body(files), 
+									 session = session)
+	
+	
+	#check to see if we've been redirected to the login page
+	if (grepl('josso/signon', r$url)) {
+		stop('Not authenticated or lack of permission to parent object\nAunthenticate with the authenticate_sb function.')
+	}
+	
+	return(as.sbitem(content(r)))
+}
+
+#' 
+#' @title Upload File to Item
+#' @description Adds a file to an item
+#'
+#' @template manipulate_item
+#' @param files A file path to upload.
+#'
+#' @return An object of class \code{sbitem}
+#'
+#' @import httr
+#'
+#' @examples 
+#' \dontrun{
+#' res <- item_create(user_id(), "testing 123")
+#' cat("foo bar", file = "foobar.txt")
+#' item_append_files(res$id, "foobar.txt")
+#' }
+#' @export
+item_append_files = function(sb_id, files, ..., session=current_session()){
+	
+	if(length(files) > 50){
+		warning('Trying to attach a large number of files to a SB item. SB imposes file limits which may cause this to fail')
+	}
+	
+	item <- as.sbitem(sb_id)
+	r = sbtools_POST(url = paste0(pkg.env$url_upload,'?id=', item$id), ...,
+									 body = multi_file_body(files), 
+									 session = session)
+  
+  return(as.sbitem(content(r)))
+	
+}
+
+multi_file_body <- function(files){
+	body = list()
+	for(i in 1:length(files)){
+		if(!file.exists(files[i])){
+			stop('This file does not exist or cannot be accessed: ', files[i])
+		}
+		
+		body[[paste0('file', i)]] = upload_file(files[i])
+	}
+	names(body) = rep('file', length(body))
+	return(body)
+}
```

Comment:
None of the code here changed, I just consolidated it into a single file so my brain didn't hurt so bad.